### PR TITLE
DOCS-330: Bucket Replication Initial Pass

### LINKED ISSUE #330: MinIO Replication
Breaking this up into parts:

1) MinIO Bucket Replication (Overview + A -> B Replication + refresh `mc replicate`)

2) MinIO Active Active replication (A <- > B)

3) MinIO `mc mirror` replication (one-time)

4) MinIO `mc mirror` active-active replication

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

Initial pass for migrating legacy bucket replication content to the new docs. partially resolves #330 . 

# Goals

- Create a landing page for MinIO Replication and get basic details into place
- Flesh out `mc admin bucket remote` and `mc replicate` commands
- Create tutorial for server-side active-passive replication

# Non-Goals

- Fully address details of how replication works
- Examples for `mc replicate`
- Additional tutorials
- Diagrams

# Build Instructions

The MinIO docs require Python 3.9+ and Node for building. For Node, use v14.5.0 or later.

1) Clone the repo, Checkout the branch (`git checkout -B DOCS-324 origin/DOCS-324`)
2) Run `npm install` from the branch
3) Create a python venv (`python -m venv venv`)
4) Install requirements (`pip install -r requirements.txt`)
5) Run `make html`
6) Run `python -m http.server --directory build/DOCS-324/html`
7) Open 127.0.0.1:8000 in your browser and take a poke around.

If you run into an error running `python -m http.server` related to `--directory`, you're python version is likely older. CD into the `build/<branch>/html` directory and run the command without the `--directory` argument.

# Key URLS

- http://127.0.0.1:8000/replication/replication-overview.html
- http://127.0.0.1:8000/replication/enable-server-side-one-way-bucket-replication.html
- http://127.0.0.1:8000/reference/minio-cli/minio-mc/mc-replicate.html#
- http://127.0.0.1:8000/reference/minio-cli/minio-mc-admin/mc-admin-bucket-remote.html#

### CODE REVIEW

File: source/reference/minio-cli/minio-mc/mc-replicate.rst [hunk 1]
```diff
@@ -15,46 +15,659 @@ Description
...
+  marker on the target bucket.
+
+- For delete operations on versions of an object,
+  MinIO replication also deletes those versions on the target bucket.
+
+MinIO requires explicitly enabling replication of delete operations using the
+:mc-cmd-option:`mc replicate add replicate` flag. This procedure includes the
+required flags for enabling replication of delete operations and delete markers.
+
+.. note::
```
Comment (ravindk89) [inline source/reference/minio-cli/minio-mc/mc-replicate.rst:196]: @ebozduman so this is not technically related to the mc rm fixups, but I figured I'd ask you what you think of this note. If this captures the core issue, I can apply it wherever we have an op that runs REMOVE under the hood.

File: source/reference/minio-cli/minio-mc/mc-replicate.rst [hunk 2]
```diff
@@ -15,46 +15,659 @@ Description
...
+destination clusters to ensure the safety of objects during transmission.
+
+MinIO does *not* support replicating client-side encrypted objects 
+(SSE-C).
 
 Examples
 --------
 
-TBD
+See the following tutorials for more complete procedures on configuring
```
Comment (ravindk89) [inline source/reference/minio-cli/minio-mc/mc-replicate.rst:226]: Will flesh this section out later.

**Review by poornas — CHANGES_REQUESTED** [review]
@ravindk89 , PR looks good - might want to touch upon how to disable replication ( by disabling a rule) and on metadata replication.

File: source/reference/minio-cli/minio-mc/mc-replicate.rst [hunk 3]
```diff
@@ -15,46 +15,659 @@ Description
...
+      :option:
+
+      *Optional*
+
+      Specify one or more comma-separated key-value pair tags which MinIO uses
+      for filtering objects to replicate. For example:
+
+      .. code-block:: shell
+
+         --tags "TAG1=VALUE,TAG2=VALUE,TAG3=VALUE"
```
Comment (poornas) [inline source/reference/minio-cli/minio-mc/mc-replicate.rst:321]: tags are delimited by &
```suggestion
         --tags "TAG1=VALUE&TAG2=VALUE&TAG3=VALUE"
```

File: source/reference/minio-cli/minio-mc/mc-replicate.rst (hunk 1 above)
Comment (ebozduman) [reply source/reference/minio-cli/minio-mc/mc-replicate.rst:196]: LGTM

Comment (ravindk89) [reply source/reference/minio-cli/minio-mc/mc-replicate.rst:196]: Great - I'll actually likely replace it with the note we spoke about on slack DM, but its good to be in the right direction :)

File: source/reference/minio-cli/minio-mc/mc-replicate.rst [hunk 4]
```diff
@@ -15,46 +15,816 @@ Description
...
 .. start-mc-replicate-desc
 
-The :mc:`mc replicate` command configures server-side bucket replication
+The :mc:`mc replicate` command configures 
+:ref:`Server-Side Bucket Replication <minio-bucket-replication-serverside>`
 between MinIO deployments. 
 
 .. end-mc-replicate-desc
 
+Create Remote Target Before Configuring Replication
```
Comment (ravindk89) [inline source/reference/minio-cli/minio-mc/mc-replicate.rst:24]: @poornas added some examples for rule modification, disable, remove.

File: source/reference/minio-cli/minio-mc/mc-replicate.rst (hunk 2 above)
Comment (ravindk89) [reply source/reference/minio-cli/minio-mc/mc-replicate.rst:253]: @poornas added examples of rule modification, removal.

File: source/replication/enable-server-side-one-way-bucket-replication.rst [hunk 5]
```diff
@@ -16,18 +16,35 @@ objects between a source and destination bucket. The destination bucket
 can be on the same MinIO cluster as the source *or* an independent remote
 MinIO cluster.
 
+The procedure on this page creates a new bucket replication rule for
```
Comment (ravindk89) [inline source/replication/enable-server-side-one-way-bucket-replication.rst:19]: @poornas quick reference to managing replication rules.

File: source/replication/replication-overview.rst [hunk 6]
```diff
@@ -37,6 +41,13 @@ server-side replication *requires* the source and destination bucket be MinIO
 clusters. The source and destination bucket *may* be the same MinIO cluster *or*
 two independent MinIO clusters.
 
+For each write operation to the bucket, MinIO checks all configured replication
```
Comment (ravindk89) [inline source/replication/replication-overview.rst:44]: @poornas very minor addition for the metadata replication. A more complete migration of the design doc will be a separate PR.

**Comment by ravindk89** [comment]
@poornas made some updates based on your feedback, PTAL

**Review by poornas — APPROVED** [review]
LGTM
